src/scion/math/HistogramTable.hpp

```diff
@@ -16,13 +16,13 @@ namespace math {
 
   /**
    *  @class
-   *  @brief Tabulated data with histogram interpolation (y is constant in x)
+   *  @brief Tabulated x,y data with histogram interpolation (y is constant in x)
    *
    *  The HistogramTable is templatised on the container type used for the
    *  x and y values in addition to the actual x and y types. This allows us to
    *  use something like utility::IteratorView instead of std::vector.
    */
-  template < typename X, typename Y = X,
+  template < typename X, typename Y,
```

Comment:
The comment might have to be adjusted here without Y=X requirement. A quick test showed this does compile and pass even with Y = X. Perhaps this indicates a test should be added to show if Y != X?

Reply:
The templates X and Y were introduced for when we would use unitised quantities (energy in eV, cross section in barns, etc.). I put the Y = X there for the case when I only wanted to use floating point values. There is no use case in which X and Y would be different types when using floating point values only (except maybe X being a floating point value and Y being an int, but I don't think that makes sense in any of the use cases we have).

As a side note: PR #75 removes all the Y = X cases and simplifies the templates somewhat.